boa/src/builtins/number/mod.rs

```diff
@@ -555,6 +555,221 @@ impl Number {
+    ///
+    /// If the number is NaN, +∞, or -∞ false is returned.
+    ///
+    /// Otherwise true is returned.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-isfinite-number
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isFinite
+    pub(crate) fn global_is_finite(
+        _this: &Value,
+        args: &[Value],
+        ctx: &mut Interpreter,
+    ) -> ResultValue {
+        if let Some(val) = args.get(0) {
+            let number = ctx.to_number(val)?;
+            Ok(Value::Boolean(number.is_finite()))
+        } else {
+            Ok(Value::Boolean(false))
+        }
+    }
+
+    /// Builtin javascript 'isNaN(number)' function.
+    ///
+    /// Converts the argument to a number, throwing a type error if the conversion is invalid.
+    ///
+    /// If the number is NaN true is returned.
+    ///
+    /// Otherwise false is returned.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-isnan-number
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN
+    pub(crate) fn global_is_nan(
+        _this: &Value,
+        args: &[Value],
+        ctx: &mut Interpreter,
+    ) -> ResultValue {
+        if let Some(val) = args.get(0) {
+            let number = ctx.to_number(val)?;
+            Ok(Value::Boolean(number.is_nan()))
+        } else {
+            Ok(Value::Boolean(true))
+        }
+    }
+
+    /// `Number.isFinite( number )`
+    ///
+    /// Checks if the argument is a number, returning false if it isn't.
+    ///
+    /// If the number is NaN, +∞, or -∞ false is returned.
+    ///
+    /// Otherwise true is returned.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-number.isfinite
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isFinite
+    pub(crate) fn number_is_finite(
+        _this: &Value,
+        args: &[Value],
+        _ctx: &mut Interpreter,
+    ) -> ResultValue {
+        Ok(Value::Boolean(if let Some(val) = args.get(0) {
+            match val {
+                Value::Integer(_) => true,
+                Value::Rational(number) => number.is_finite(),
+                Value::Object(ref obj) => {
+                    let obj = obj.borrow();
+                    if let Some(number) = obj.as_number() {
+                        number.is_finite()
+                    } else {
+                        false
+                    }
+                }
+                _ => false,
+            }
+        } else {
+            false
+        }))
```

Comment:
```suggestion
        if let Some(Value::Boolean(number)) = args.get(0) {
            return Ok(number.is_finite().into());
        }
        
        Ok(Value::from(false))
```
The `Number.isFinite()` does not check for number object <https://tc39.es/ecma262/#sec-number.isfinite>

Reply:
I might be missing something in the spec, but from what I can see it should consider Number objects. In both chrome and firefox the following snippets are valid:
```javascript
Number.isFinite(Number(5)) // true
Number.isFinite(Number(Infinity)) // false
```

Reply:
> I might be missing something in the spec, but from what I can see it should consider Number objects. In both chrome and firefox the following snippets are valid:
> 
> ```js
> Number.isFinite(Number(5)) // true
> Number.isFinite(Number(Infinity)) // false
> ```

The `Number` function has two sides:
```js
Number(10) // 10 number primitive is returned.
new Number(10) // `Number { 10 }` number object is returned with value 10
```

Reply:
Ah, OK, I wasn't aware of the difference. I'll make those changes then.